packages/blade/src/components/BaseMenu/BaseMenuItem.tsx

```diff
@@ -0,0 +1,168 @@
+    width: '100%',
+    [`@media ${getMediaQuery({ min: props.theme.breakpoints.m })}`]: {
+      padding: makeSize(props.theme.spacing[3]),
+    },
+    '&:focus-visible': getFocusRingStyles({ theme: props.theme }),
+    '&:hover:not([aria-disabled=true]), &.has-submenu-open': {
+      backgroundColor:
+        props.color === 'negative'
+          ? props.theme.colors.interactive.background.negative.faded
+          : props.theme.colors.interactive.background.gray.default,
+    },
+    '&[aria-selected=true]': {
+      backgroundColor: props.theme.colors.interactive.background.primary.faded,
+    },
+    '&[aria-selected=true]:hover': {
+      backgroundColor: props.theme.colors.interactive.background.primary.fadedHighlighted,
+    },
+  };
+});
+
+type BaseMenuItemProps = {
+  as?: 'a' | 'button';
+  title: string;
+  description?: string;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+  titleSuffix?: React.ReactNode;
+  selectionType?: 'single' | 'multiple';
+  color?: 'negative';
+  className?: string;
+  href?: string;
+};
+
+const menuItemTitleColor = {
+  negative: {
+    default: 'feedback.text.negative.intense',
+    disabled: 'interactive.text.gray.disabled',
+  },
+  normal: { default: 'interactive.text.gray.normal', disabled: 'interactive.text.gray.disabled' },
+} as const;
+
+const menuItemDescriptionColor = {
+  default: 'interactive.text.gray.muted',
+  disabled: 'interactive.text.gray.disabled',
+} as const;
+
+const _BaseMenuItem: React.ForwardRefRenderFunction<BladeElementRef, BaseMenuItemProps> = (
+  {
+    title,
+    description,
+    as,
+    leading,
+    trailing,
+    titleSuffix,
+    isDisabled,
+    selectionType,
+    isSelected,
+    color,
+    ...props
+  },
+  ref,
+): React.ReactElement => {
+  return (
+    <StyledMenuItemContainer
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref={ref as any}
+      as={as}
+      type={as === 'button' ? 'button' : undefined}
+      disabled={isDisabled}
+      {...makeAccessible({
+        role: 'menuitem',
+        disabled: isDisabled,
+        selected: isSelected,
+      })}
+      color={color}
+      {...props}
+    >
+      <Box
+        display="flex"
+        justifyContent="center"
+        alignItems="center"
+        maxHeight={makeSize(size[20])}
+        width="100%"
+      >
+        <Box display="flex" justifyContent="center" alignItems="center">
+          {selectionType === 'multiple' ? (
+            // Adding aria-hidden because the listbox item in multiselect in itself explains the behaviour so announcing checkbox is unneccesary and just a nice UI tweak for us
+            <BaseBox
+              pointerEvents="none"
+              paddingRight="spacing.2"
+              {...makeAccessible({
+                hidden: true,
+              })}
+            >
+              <Checkbox isChecked={isSelected} tabIndex={-1} isDisabled={isDisabled}>
```

Comment:
Why do we need a checkbox?

Reply:
Been trying to have this component that covers both MenuItem and ActionList styling. Will use this in ActionListItem as well where we need checkbox

Reply:
Did the refactor to use this in ActionListItem as well #2232